docs - update caching (#29401)
metabase/metabase: docs/configuring-metabase/caching.md

# Caching query results

If your results don't change frequently, you may want to cache your results, that is: store your results in Metabase so that the next time you visit the question, Metabase can retrieve the stored results rather than query the database again. 

> For example, if your data only updates once a day, there's no point in querying the database more than once a day, as they data won't have changed. Returning cached results can be significantly faster, as the database won't have to redo the work to answer your query.

Reviewer: ```suggestion
For example, if your data only updates once a day, there's no point in querying the database more than once a day, as they data won't have changed. Returning cached results can be significantly faster, as the database won't have to recompute the results to answer your query.
```

Metabase gives you the ability to automatically cache query results. On [paid plans](https://www.metabase.com/pricing/), you can also control caching for specific:

- [Databases](#caching-per-database)
- [Questions](#caching-per-question)
- [Dashboards](#caching-per-dashboard)

> You can also [cache models](../data-modeling/models.md#caching-individual-models).

Reviewer: Why is model caching excluded/separate from the list above?

Not for this PR, just noting for myself: how does caching work with actions? Will enabling model actions disable caching for models with actions? (I can follow up with a PR to answer this, if necessary).

## Enabling global caching

> 1. Go to **Admin settings** > **Caching** (in the side nav).

Reviewer: ```suggestion
1. Go to **Admin settings** > **Caching** (in the sidebar nav).
```

2. Click the toggle under **Saved Questions**.

Once you've enabled caching, you can choose when and what to cache from your [caching settings](#caching-settings).

> By default, questions will get cached once their [average execution time](#average-query-execution-time) meets a [minimum query duration](#minimum-query-duration) of 60 seconds.

Reviewer: ```suggestion
By default, questions will get cached once their [average execution time](#average-query-execution-time) meets a [minimum query duration](#minimum-query-duration) of 60 seconds. You can change this minimum query duration in the [caching settings](#caching-settings).
```

For example, if you enter a TTL multiplier of 10, a query that takes 5 seconds on average will be cached for 50 seconds. A query that takes 10 minutes will be cached for 100 minutes. This way, each query’s cache lifetime is proportional to that query's execution time.

### Max cache entry size

To prevent cached results from taking up too much space on your server, you can set the maximum size of each question's cache in kilobytes.

## Advanced caching controls

{% include plans-blockquote.html feature="Advanced caching controls" %}

All Metabase editions include global caching controls. On paid plans, you can override your global [time-to-live (TTL) setting](#cache-time-to-live-ttl-multiplier) to set different cache lifetimes for specific databases, questions, or dashboards.

### Caching per database

This setting tells Metabase how long to keep the cached results from a specific database.

1. Make sure [caching is enabled](#enabling-global-caching).
2. Go to **Admin settings** > **Databases** and select your database.
3. Open **Advanced options** and find the **Default result cache duration**.
4. Click **Custom** and enter a cache duration in hours.

> This setting is useful for databases that take longer to query, or databases that are kept up to date on a special cadence. 

Reviewer: ```suggestion
The cache duration setting is useful for databases that take longer to query, or databases that are kept up to date on a special cadence. 
```

This setting will override your [global TTL settings](#cache-time-to-live-ttl-multiplier).

### Caching per question

> This setting tells Metabase how long to keep the cached results for a specific question. You'll only find these cache settings on questions that exceed the [minimum query duration](#minimum-query-duration).

Reviewer: ```suggestion
You can tell Metabase how long to keep the cached results for specific questions. You'll only find these cache settings on questions that exceed the [minimum query duration](#minimum-query-duration).
```

1. Make sure [caching is enabled](#enabling-global-caching).
2. Go to your question.
3. Click on the **info** icon.
4. Click **Cache configuration**.
5. Enter a cache duration in hours.
6. Click **Save changes**.

You can use this setting to update questions on the same cadence as your data. For example, if your data gets updated daily, you can set the **Cache configuration** to 24 hours.

If set, your question cache duration will override:

- [global cache duration](#cache-time-to-live-ttl-multiplier)
- [database cache duration](#caching-per-database)

### Caching per dashboard

> This setting tells Metabase how long to keep the cached results for each of the questions on a dashboard.

Reviewer: ```suggestion
You can tell Metabase how long to keep the cached results for each of the questions on a dashboard.
```

Editing these slightly as these aren't really single-location settings, like in the admin panel; "these settings" isn't quite right; and the rephrasing makes it seem like you can communicate with Metabase.
